// material-ui-image/lib/image.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Image = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _materialUi = require('material-ui');

var _colors = require('material-ui/styles/colors');

var colors = _interopRequireWildcard(_colors);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

function getRandomColor() {
  var colorNames = Object.keys(colors);
  return colors[colorNames[Math.floor(colorNames.length * Math.random())]];
}

var styles = {
  root: {
    width: 300,
    height: 200
  },
  loading: {
    position: 'relative'
  },
  img: {
    width: 'inherit',
    height: 'inherit'
  }
};

var Image = exports.Image = function (_Component) {
  _inherits(Image, _Component);

  function Image(props) {
    _classCallCheck(this, Image);

    var _this = _possibleConstructorReturn(this, (Image.__proto__ || Object.getPrototypeOf(Image)).call(this, props));

    _this.state = {
      color: getRandomColor()
    };
    return _this;
  }

  _createClass(Image, [{
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      if (nextProps.src !== this.props.src) {
        this.setState({
          color: getRandomColor()
        });
      }
    }
  }, {
    key: 'render',
    value: function render() {
      var _this2 = this;

      var _props = this.props;
      var color = _props.color;
      var disableSpinner = _props.disableSpinner;
      var imageStyle = _props.imageStyle;
      var src = _props.src;
      var style = _props.style;
      var loadingSize = _props.loadingSize;
      var loadingStyle = _props.loadingStyle;


      return _react2.default.createElement(
        'div',
        { style: _extends({}, styles.root, { backgroundColor: color || this.state.color }, style) },
        !disableSpinner && !this.state.imageLoaded && !this.state.imageError ? _react2.default.createElement(_materialUi.RefreshIndicator, {
          size: loadingSize,
          left: style && style.width ? style.width / 2 - loadingSize : styles.root.width / 2 - loadingSize / 2,
          top: style && style.height ? style.height / 2 - loadingSize : styles.root.height / 2 - loadingSize / 2,
          status: 'loading',
          style: _extends({}, styles.loading, loadingStyle)
        }) : null,
        src ? _react2.default.createElement('img', _extends({}, this.props, {
          onClick: this.props.onTouchTap,
          style: _extends({}, styles.img, { opacity: !this.state.imageLoaded ? 0 : 1, transition: 'all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)' }, imageStyle),
          onLoad: function onLoad() {
            return _this2.setState({ imageLoaded: true });
          },
          onError: function onError() {
            return _this2.setState({ imageError: true });
          }
        })) : null
      );
    }
  }]);

  return Image;
}(_react.Component);

Image.defaultProps = {
  disableSpinner: false,
  loadingSize: 40
};

Image.propTypes = {
  src: _react.PropTypes.string.isRequired,
  color: _react.PropTypes.string,
  disableSpinner: _react.PropTypes.bool,
  onTouchTap: _react.PropTypes.func,
  style: _react.PropTypes.object,
  imageStyle: _react.PropTypes.object,
  loadingSize: _react.PropTypes.number,
  loadingStyle: _react.PropTypes.object
};
